lib/dateTime.js

```diff
@@ -216,16 +308,82 @@ function formatDateString(date) {
+			const mins = match[5] ? prePadByZero(match[5], 2) : '00';
+			dateString = `${date.year}-${date.month}-${date.date}T${date.hours}:${date.minutes}:${date.seconds}${match[2]}${prePadByZero(match[3], 2)}:${mins}`;
 		}
 	}
 	return dateString;
 }
 
+function getTimezoneFromDuplicatedAbbreviation(abbrTimezone) {
+	const longTimezone = getDocumentLocaleSettings().timezone.identifier;
+	switch (abbrTimezone) {
+		case 'ADT':
+			// Arabia Daylight Time +0400 (Asia)
+			// Atlantic Daylight Time -0300 (America)
+			return longTimezone.includes('America') ? '-0300' : '+0400';
+		case 'AMT':
+			// Amazon Time (Brazil) -0400 (America)
+			// Armenia Time +0400 (Asia)
+			return longTimezone.includes('America') ? '-0400' : '+0400';
+		case 'AMST':
+			// Amazon Summer Time -0300 (America)
+			// Armenia Summer Time +0500 (Asia)
+			return longTimezone.includes('America') ? '-0300' : '+0500';
+		case 'AST':
+			// Atlantic Standard Time -0400 (America)
+			// Arabia Standard Time +0300 (Asia)
+			return longTimezone.includes('America') ? '-0400' : '+0300';
+		case 'BST':
+			// Bangladesh Standard Time +0600 (Asia)
+			// British Summer Time +0100 (Europe)
+			// Bougainville Standard Time +1100 (Pacific)
+			if (longTimezone.includes('Asia')) return '+0600';
+			else if (longTimezone.includes('Pacific')) return '+1100';
+			else return '+0100';
+		case 'CDT':
+			// Central Daylight Time -0500 (America)
+			// Cuba Daylight Time -0400 (America/Havana)
+			return longTimezone === 'America/Havana' ? '-0400' : '-0500';
+		case 'CST':
+			// Cuba Standard Time -0500 (America/Havana)
+			// Central Standard Time -0600 (America)
+			// China Standard Time +0800 (Asia)
+			if (longTimezone === 'America/Havana') return '-0500';
+			else if (longTimezone.includes('America')) return '-0600';
+			else return '+0800';
+		case 'GST':
+			// South Georgia and the South Sandwich Islands Time -0200 (Atlantic)
+			// Gulf Standard Time +0400 (Asia)
+			return longTimezone.includes('EurAtlanticope') ? '-0200' : '+0400';
+		case 'IST':
+			// Indian Standard Time +0530 (Asia/Calcutta & Asia/Colombo)
+			// Irish Standard Time +0100 (Europe)
+			// Israel Standard Time +0200 (Asia)
+			if (longTimezone === 'Asia/Calcutta' || longTimezone === 'Asia/Colombo') return '+0530';
+			else if (longTimezone.includes('Asia')) return '+0200';
+			else return '+0100';
+		case 'MST':
+			// Malaysia Standard Time +0800 (Asia)
+			// Mountain Standard Time -0700 (America)
+			return longTimezone.includes('America') ? '-0700' : '+0800';
+		case 'PYT':
+			// Paraguary Time -0400 (South America)
+			// Pyongyan Time +0830 (Asia)
+			return longTimezone.includes('Asia') ? '+0830' : '-0400';
+		case 'WST':
+			// West Samoa Time +1400 (Pacific)
+			// Western Sahara Summer Time +0100 (Africa)
+			return longTimezone.includes('Pacific') ? '+1400' : '+0100';
+		default:
+			throw new Error(`Invalid timezone: unable to retrieve timezone offset for ${longTimezone}`);
```

Comment:
Are we reasonably confident that this isn't going to throw very often? Like I guess previously this would have resulted in an `undefined` timezone which might have blown up elsewhere in the code.

Reply:
It shouldn't blow up anymore than before. Before if the timezone was `undefined` then it would fail in `getDateStringWithTimezone` slightly later in the same function so I figured adding this would help make debugging more clear. I think throwing somewhere in here is probably what we want since we wouldn't want `convertLocalToUTCDateTime` to return an incorrect value.

Reply:
Sounds good. Yeah at least now we could actively search for this error in the wild!